File: lesson-2_recursion-and-recurrence/fibonacci/recursion/main.c
#include <stdio.h>

int fibonacci(int);

int main(void)
{
    int n;
    scanf("%d", &n);
    printf("%d\n", fibonacci(n));
    return 0;
}

int fibonacci(int n)
{
    if (n < 2)
    {
        return 1;
    }
    return fibonacci(n - 1) + fibonacci(n - 2);
}

File: lesson-2_recursion-and-recurrence/fibonacci/recursion/main.js
'use strict'

const readline = require('readline')
const rl = readline.createInterface({ input: process.stdin })

rl.on('line', input => {
    console.log(fibonacci(input))
    rl.close()
})

function fibonacci(n) {
    if (n < 2) {
        return 1
    }
    return fibonacci(n - 1) + fibonacci(n - 2)
}
